lib/api/api-request.js

```diff
@@ -111,12 +111,13 @@ class RequestHandler extends AsyncResource {
 
     const parsedHeaders = responseHeaders === 'raw' ? util.parseHeaders(rawHeaders) : headers
     const contentType = parsedHeaders['content-type']
-    const contentLength = parsedHeaders['content-length']
     const body = new Readable({
       resume,
       abort,
       contentType,
-      contentLength: this.method === 'HEAD' ? null : contentLength,
+      contentLength: this.method !== 'HEAD' && parsedHeaders['content-length']
+        ? Number(parsedHeaders['content-length'])
```

Comment:
is it ensured, that it is a positive integer?

Reply:
No, but does it matter?

Reply:
what happens if it is NaN?

Reply:
If the server is fully compliant, it should never sent any non-positive value; if a non-positive value is sent, the parser will fail with a non-compliant error